tests/chainer_tests/functions_tests/math_tests/test_zeta.py

```diff
@@ -0,0 +1,76 @@
+import unittest
+
+import numpy
+
+import chainer
+import chainer.functions as F
+from chainer import testing
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'shape': [(), (5, 10)]
+}))
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0', 'cuda:1']
+    })
+)
+@testing.with_requires('scipy')
+class TestZeta(testing.FunctionTestCase):
+
+    def setUp(self):
+        self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-3}
+        self.check_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+        self.check_double_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+
+        self._x = numpy.random.uniform(self.shape).astype(self.dtype)
+
+    def generate_inputs(self):
+        q = numpy.random.uniform(self.shape).astype(self.dtype)
+        return q,
+
+    def forward_expected(self, inputs):
+        q, = inputs
+        import scipy
+        y_expect = scipy.special.zeta(self._x, q)
+        return y_expect.astype(self.dtype),
+
+    def forward(self, inputs, device):
+        q, = inputs
+        y = F.zeta(device.send(self._x.astype(q.dtype)), q)
+        return y,
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'shape': [(), (5, 10)]
+}))
+@testing.without_requires('scipy')
+class TestZetaExceptions(unittest.TestCase):
+    def setUp(self):
+        self.q = numpy.random.uniform(self.shape).astype(self._x.dtype)
```

Comment:
Another one, although I'm not really sure you need to parameterize this test at all.
```suggestion
        self.q = numpy.random.uniform(self.shape).astype(self.dtype)
```

Reply:
I'm not sure how this change would impact anything apart from maintaining consistency. I have added it :+1:

Reply:
`self` is never assigned `_x`.